packages/plugins/plugin-table/src/next/hooks/useTable.ts

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { effect } from '@preact/signals-core';
+import { type RefObject, useCallback, useEffect, useMemo } from 'react';
+
+import { type DxGridElement, type DxGridAxisMeta, type DxGridCells } from '@dxos/react-ui-grid';
+
+import { type TableEvent, type ColumnDefinition, createTable, updateTable } from '../table';
+
+// TODO(Zan): Take ordering here (or order based on some stored property).
+// When the order changes, we should notify the consumer.
+export const useTable = (columnDefinitions: ColumnDefinition[], data: any[], gridRef: RefObject<DxGridElement>) => {
+  const table = useMemo(() => createTable(columnDefinitions, data), [columnDefinitions, data]);
```

Comment:
the seems to create a datastructure that should be passed in from external state.
i.e., data here should be the already reactive result of a query.